l1-contracts/test/foundry/unit/concrete/state-transition/chain-deps/facets/Mailbox/RequestL2Transaction.t.sol

```diff
@@ -0,0 +1,130 @@
+            _l2Value: 0,
+            _calldata: tempBytes,
+            _l2GasLimit: 0,
+            _l2GasPerPubdataByteLimit: 0,
+            _factoryDeps: tempBytesArr,
+            _refundRecipient: tempAddress
+        });
+    }
+
+    function test_RevertWhen_wrongL2GasPerPubdataByteLimit() public {
+        vm.expectRevert(bytes("qp"));
+        mailboxFacet.requestL2Transaction({
+            _contractL2: tempAddress,
+            _l2Value: 0,
+            _calldata: tempBytes,
+            _l2GasLimit: 0,
+            _l2GasPerPubdataByteLimit: 0,
+            _factoryDeps: tempBytesArr,
+            _refundRecipient: tempAddress
+        });
+    }
+
+    function test_RevertWhen_msgValueDoesntCoverTx() public {
+        utilsFacet.util_setBaseTokenGasPriceMultiplierDenominator(1);
+        tempBytesArr = new bytes[](1);
+
+        uint256 baseCost = mailboxFacet.l2TransactionBaseCost(10000000, 1000000, REQUIRED_L2_GAS_PRICE_PER_PUBDATA);
+        uint256 l2Value = 1 ether;
+        uint256 mintValue = baseCost + l2Value;
+
+        vm.expectRevert(bytes("mv"));
+        mailboxFacet.requestL2Transaction{value: mintValue - 1}({
+            _contractL2: tempAddress,
+            _l2Value: l2Value,
+            _calldata: tempBytes,
+            _l2GasLimit: 1000000,
+            _l2GasPerPubdataByteLimit: REQUIRED_L2_GAS_PRICE_PER_PUBDATA,
+            _factoryDeps: tempBytesArr,
+            _refundRecipient: tempAddress
+        });
+    }
+
+    function test_RevertWhen_factoryDepsLengthExceeded() public {
+        tempBytesArr = new bytes[](MAX_NEW_FACTORY_DEPS + 1);
+
+        vm.expectRevert(bytes("uj"));
+        mailboxFacet.requestL2Transaction({
+            _contractL2: tempAddress,
+            _l2Value: 0,
+            _calldata: tempBytes,
+            _l2GasLimit: 0,
+            _l2GasPerPubdataByteLimit: REQUIRED_L2_GAS_PRICE_PER_PUBDATA,
+            _factoryDeps: tempBytesArr,
+            _refundRecipient: tempAddress
+        });
+    }
+
+    function test_success_requestL2Transaction() public {
+        utilsFacet.util_setBaseTokenGasPriceMultiplierDenominator(1);
+        utilsFacet.util_setPriorityTxMaxGasLimit(100000000);
+
+        bytes[] memory factoryDeps = new bytes[](1);
+        factoryDeps[0] = "11111111111111111111111111111111";
+
+        uint256 baseCost = mailboxFacet.l2TransactionBaseCost(10000000, 1000000, REQUIRED_L2_GAS_PRICE_PER_PUBDATA);
+        uint256 l2Value = 1 ether;
+        uint256 mintValue = baseCost + l2Value;
+
+        address baseTokenBridge = makeAddr("baseTokenBridge");
+        utilsFacet.util_setBaseTokenBridge(baseTokenBridge);
+
+        vm.mockCall(
+            baseTokenBridge,
+            abi.encodeWithSelector(
+                IL1SharedBridge.bridgehubDepositBaseToken.selector,
+                eraChainId,
+                sender,
+                ETH_TOKEN_ADDRESS,
+                mintValue
+            ),
+            ""
+        );
```

Comment:
Please try to avoid the use of `mockCall` at all. What you are doing here can be done by calling a `DummyL1SharedBridge.bridgehubDepositBaseToken`. 

Also, try this function in the states when L1SharedBridge is paused and unpaused.
Also, please check for the change in balances in the contract and something like `chainBalance[_chainId][_l1Token]` in the DummyL1SharedBridge.

Reply:
fixed